app/src/app/views/stix/collection/collection-import/collection-import-workflow/collection-import.component.ts

```diff
@@ -249,6 +257,56 @@ export class CollectionImportComponent implements OnInit {
         })
     }
 
+    /**
+     * Determine if the user can force import a collection after the post collection
+     * call fails. Users may force an import when:
+     *   1. the collection bundle is already in the database (user is re-importing the bundle)
+     *   2. the collection bundle has one or more objects with an invalid ATT&CK spec version.
+     * @param error the resulting error from previewing the collection
+     * @returns true if the user can force import the collection; false otherwise
+     */
+    private canForceImport(err: any): boolean {
+        if (!err.error) return false;
+        if (err.error.bundleErrors.duplicateCollection ||
+            err.error.objectErrors.summary.invalidAttackSpecVersionCount) return true;
+        return false;
+    }
+
+    /**
+     * Set up the objects to force import from a collection
+     * @param collectionBundle the collection bundle to import
+     * @param error the resulting error from previewing the collection
+     */
+    public errorPreview(collectionBundle: any, error: any): void {
+        this.import_error = error;
+        this.collectionBundle = collectionBundle; //save for later
+
+        let invalidObjects: string[] = error.objectErrors.errors.map(obj => obj.id);
+        let selected: string[] = [];
+        for (let object of this.collectionBundle.objects) {
+            if ("type" in object && object.type == 'x-mitre-collection') continue;
+            if ("id" in object && !invalidObjects.includes(object.id)) {
+                selected.push(object.id);
+            }
+        }
```

Comment:
Useful addition, though you might want to make it clearer to the user that the invalid objects have been deselected.

Reply:
Actually, I'm not sure this is a good addition. The post-import step should give a list of objects that couldn't be imported (that had errors). If they are excluded via this means, the user has no way of finding out what they are. So it may be better to let those objects fail so the later error handling can report them. 

The list of errors also ends up in the database that way, which is useful for later investigation. In this implementation the avoidance of saving those objects makes it so it's never permanently recorded anywhere.